### PR TITLE
fix(server): add missing file extensions to library files

### PR DESCRIPTION
**Pull request by etnoy**
* Detect when the basename is invalid and force refresh
* Actually re-import new filename upon refresh

This adds to #8083 and #8052 and makes a retroactive fix for all affected assets

### CODE REVIEW

File: server/src/services/library.service.ts [hunk 1]
```diff
@@ -521,6 +529,8 @@ export class LibraryService extends EventEmitter {
       await this.assetRepository.updateAll([existingAssetEntity.id], {
         fileCreatedAt: stats.mtime,
         fileModifiedAt: stats.mtime,
+        localDateTime: stats.mtime,
```
Comment (jrasm91) [inline server/src/services/library.service.ts:532]: Does this need to be offset so it's in utc?

**Review by jrasm91 — APPROVED** [review]
Nice!

File: server/src/services/library.service.ts (hunk 1 above)
Comment (etnoy) [reply server/src/services/library.service.ts:532]: hm...good point. Maybe let's not touch that rabbit hole today

File: server/src/services/library.service.ts [hunk 2]
```diff
@@ -453,6 +453,20 @@ export class LibraryService extends EventEmitter {
...
+    } else if (existingAssetEntity.originalFileName !== parse(assetPath).base) {
+      // Asset base filename is invalid.
+      // This cleans up an old bug where the file extension wasn't handled properly in previous versions
+      // We can likely remove this check in the second half of 2024
+      console.warn(
+        assetPath
+      );
+      console.warn(
+        existingAssetEntity.originalFileName
+      );
```
Comment (mertalev) [inline server/src/services/library.service.ts]: Nit: remove console statements.

File: server/src/services/library.service.ts [hunk 3]
```diff
@@ -453,6 +453,14 @@ export class LibraryService extends EventEmitter {
         `File modification time has changed, re-importing asset: ${assetPath}. Old mtime: ${existingAssetEntity.fileModifiedAt}. New mtime: ${stats.mtime}`,
       );
       doRefresh = true;
+    } else if (existingAssetEntity.originalFileName !== parse(assetPath).base) {
```
Comment (mertalev) [inline server/src/services/library.service.ts:456]: Nit: the asset path is parsed twice. You can move it to a variable to reuse it.

File: server/src/services/library.service.ts [hunk 4]
```diff
@@ -453,6 +453,14 @@ export class LibraryService extends EventEmitter {
...
+      // Asset base filename is invalid.
+      // This cleans up an old bug where the file extension wasn't handled properly in previous versions
+      // We can likely remove this check in the second half of 2024
```
Comment (mertalev) [inline server/src/services/library.service.ts:459]: This should be a TODO

File: server/src/services/library.service.ts [hunk 5]
```diff
@@ -453,6 +453,14 @@ export class LibraryService extends EventEmitter {
         `File modification time has changed, re-importing asset: ${assetPath}. Old mtime: ${existingAssetEntity.fileModifiedAt}. New mtime: ${stats.mtime}`,
       );
       doRefresh = true;
+    } else if (existingAssetEntity.originalFileName !== parse(assetPath).base) {
+      // Asset base filename is invalid.
```
Comment (mertalev) [inline server/src/services/library.service.ts:457]: Nit: redundant comment.

File: server/src/services/library.service.ts [hunk 6]
```diff
@@ -453,6 +453,14 @@ export class LibraryService extends EventEmitter {
...
+      // This cleans up an old bug where the file extension wasn't handled properly in previous versions
```
Comment (mertalev) [inline server/src/services/library.service.ts:458]: Nit: this is a bit verbose

**Review by mertalev — APPROVED** [review]

File: server/src/services/library.service.ts (hunk 4 above)
Comment (etnoy) [reply server/src/services/library.service.ts:459]: Good point!

File: server/src/services/library.service.ts (hunk 2 above)
Comment (etnoy) [reply server/src/services/library.service.ts]: Done

File: server/src/services/library.service.ts (hunk 3 above)
Comment (etnoy) [reply server/src/services/library.service.ts:456]: Done, thanks

File: server/src/services/library.service.ts (hunk 6 above)
Comment (etnoy) [reply server/src/services/library.service.ts:458]: Done

File: server/src/services/library.service.ts (hunk 5 above)
Comment (etnoy) [reply server/src/services/library.service.ts:457]: Done

**Review by mertalev — APPROVED** [review]

**Pull request merged**